gtsfm/loader/astronet_loader.py

```diff
@@ -0,0 +1,231 @@
+
+Authors: Travis Driver
+"""
+
+import os
+from pathlib import Path
+from typing import Optional, Dict, Tuple, List
+
+from gtsam import Cal3Bundler, Pose3, Rot3, Point3, SfmTrack
+
+import gtsfm.utils.io as io_utils
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+from thirdparty.colmap.scripts.python.read_write_model import read_model
+from thirdparty.colmap.scripts.python.read_write_model import Camera as ColmapCamera
+from thirdparty.colmap.scripts.python.read_write_model import Image as ColmapImage
+from thirdparty.colmap.scripts.python.read_write_model import Point3D as ColmapPoint3D
+
+
+logger = logger_utils.get_logger()
+
+
+class AstroNetLoader(LoaderBase):
+    """Simple loader class that reads an AstroNet data segment.
+
+    Refs:
+    - https://github.com/travisdriver/astronet
+    """
+
+    def __init__(
+        self,
+        data_dir: str,
+        use_gt_extrinsics: bool = True,
+        use_gt_sfmtracks: bool = False,
+        max_frame_lookahead: int = 2,
+    ) -> None:
+        """Initialize loader from a specified segment directory (data_dir) on disk.
+
+        <data_dir>/
+             ├── images/: undistorted grayscale images
+             ├── cameras.bin: camera calibrations (see https://colmap.github.io/format.html#cameras-txt)
+             ├── images.bin: 3D poses and 2D tracks (see https://colmap.github.io/format.html#images-txt)
+             └── points3D.bin: 3D tracks (see https://colmap.github.io/format.html#points3d-txt)
+
+
+        Args:
+            data_dir: path to directory containing the COLMAP-formatted data: cameras.bin, images.bin, and points3D.bin
+            gt_scene_mesh_path (optional): path to file of target small body surface mesh.
+                Note: vertex size mismath observed when reading in from OBJ format. Prefer PLY.
+            use_gt_intrinsics: whether to use ground truth intrinsics. If calibration is
+               not found on disk, then use_gt_intrinsics will be set to false automatically.
+            use_gt_extrinsics (optional): whether to use ground truth extrinsics.
+            use_gt_sfmtracks (optional): whether to use ground truth tracks.
+            max_frame_lookahead (optional): maximum number of consecutive frames to consider for
+                matching/co-visibility. Any value of max_frame_lookahead less than the size of
+                the dataset assumes data is sequentially captured.
+
+        Raises:
+            FileNotFoundError if image path does not exist.
+            RuntimeError if ground truth camera calibrations not provided.
+        """
+        self._use_gt_extrinsics = use_gt_extrinsics
+        self._use_gt_sfmtracks = use_gt_sfmtracks
+        self._max_frame_lookahead = max_frame_lookahead
+
+        # Use COLMAP model reader to load data and convert to GTSfM format.
+        if Path(data_dir).exists():
```

Comment:
what's the expected behavior if `data_dir` does not exist?

Reply:
Added `FileNotFound` exception